### ISSUE TITLE
Create Upgrade Documentation

### ISSUE THREAD

**Issue by AlexNPavel** [operator-framework/operator-sdk#947]
Upgrading the projects generated by the SDK is important to make sure that operators are using the latest features of the SDK and to make sure the operators/projects are compatible with the latest SDK binaries. For most non-breaking changes, it should be simple enough to use the `operator-sdk print-deps` command, but it would be good to have a small doc talking about that that we can point users to.

**Comment by wreed4** [comment]
Dealing with this now, so question... Is there ever a case where the code generated by the operator-sdk, the scaffolding itself,  needs to be updated when updating operator-sdk version?? And if so, what is the process for that?  This issue mentions simply the dependency side of the upgrade, but what should a user do if the actual code generated needs to change?  Or is that not something that happens?

**Comment by estroz** [comment]
@wreed4 in most cases no if you're migrating between two minor versions, ex. v0.1.0 to v0.2.0. However there are some changes required. @hasbro17 is working on a migration doc for each minor version bump. The breaking changes can be found in our [changelog](https://github.com/operator-framework/operator-sdk/blob/master/CHANGELOG.md), which is informative but hard to parse.

**Comment by nitishm** [comment]
@estroz So you are saying that the generated scaffold for v0.8.0 is most likely the same as v0.9.0 (or any minor upgrade) for the most part ? We shouldnt worry about updating the scaffolding when we bump up our operator-sdk binary to a higher release ?

**Comment by estroz** [comment]
@nitishm you likely won't have to, but while we're sub-1.0 we may break the scaffold in a minor release. We will make sure to list these as **Breaking Changes** in the release notes and have a migration guide ready.

That being said, there may be other changes made to what the SDK binary expects from your operator, outside of its project scaffold. These will also be listed as breaking changes in the changelog.
